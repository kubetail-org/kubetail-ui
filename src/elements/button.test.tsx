import { render } from '@testing-library/react';

import { Button } from './button';
import type { ButtonVariantProps } from './button';

describe('Button', () => {
  it('renders contents properly', () => {
    const { asFragment } = render(<Button>My Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders variants properly', () => {
    ['primary', 'secondary', 'outline', 'ghost', 'destructive', 'link'].forEach((variant) => {
      const { asFragment } = render(<Button variant={variant as ButtonVariantProps['variant']} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders sizes properly', () => {
    ['xs', 'sm', 'md', 'lg', 'xl'].forEach((size) => {
      const { asFragment } = render(<Button size={size as ButtonVariantProps['size']}>My Button</Button>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders disabled=true properly', () => {
    const { asFragment } = render(<Button disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('appends custom className', () => {
    const { asFragment } = render(<Button className="my-extra-class" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
