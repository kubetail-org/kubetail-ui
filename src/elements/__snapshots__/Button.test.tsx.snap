// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > appends custom className 1`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md my-extra-class"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button > renders as input tag properly 1`] = `
<DocumentFragment>
  <input
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md"
    role="button"
    tabindex="0"
    type="submit"
    value="My Button"
  />
</DocumentFragment>
`;

exports[`Button > renders as link when href is set 1`] = `
<DocumentFragment>
  <a
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md"
    href="#"
    role="button"
    tabindex="0"
  />
</DocumentFragment>
`;

exports[`Button > renders contents properly 1`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;

exports[`Button > renders disabled=true properly 1`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md"
    disabled=""
    type="button"
  />
</DocumentFragment>
`;

exports[`Button > renders sizes properly 1`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-2.5 py-1.5 text-xs rounded"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;

exports[`Button > renders sizes properly 2`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-3 py-2 text-xs rounded-md"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;

exports[`Button > renders sizes properly 3`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;

exports[`Button > renders sizes properly 4`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-base rounded-md"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;

exports[`Button > renders sizes properly 5`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-6 py-3 text-base rounded-md"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;

exports[`Button > renders variants properly 1`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 shadow-sm px-4 py-2 text-sm rounded-md"
    intent="primary"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button > renders variants properly 2`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/60 shadow-sm px-4 py-2 text-sm rounded-md"
    intent="secondary"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button > renders variants properly 3`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 px-4 py-2 text-sm rounded-md"
    intent="white"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button > renders variants properly 4`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-danger text-danger-foreground hover:bg-danger/90 shadow-sm px-4 py-2 text-sm rounded-md"
    intent="danger"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button > renders variants properly 5`] = `
<DocumentFragment>
  <button
    class="uppercase inline-flex items-center justify-center whitespace-nowrap font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 px-4 py-2 text-sm rounded-md"
    intent="success"
    type="button"
  />
</DocumentFragment>
`;
